manifests/global.pp

```diff
@@ -118,13 +101,29 @@
     }
   }
 
+  $effective_hostname = $hostname ? {
+    String  => $hostname,
+    default => $::networking['fqdn'],
+  }
```

Comment:
This is ugly, but was the only way to preserve the current behavior:
- When called with a ``$hostname``, it is used both by ``hostnamectl`` and the template
- When called without a ``$hostname``, the template gets a value through facts, but the ``hostnamectl`` ``exec`` isn't evaluated; I know the result is the same, but to simplify the code, we could simply pass ``$::networking['fqdn']`` as default class parameter, and let the ``unless`` attribute of the ``exec`` do its job.

What do you think?